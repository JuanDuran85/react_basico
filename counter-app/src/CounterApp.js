import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CountApp = ({value}) => {

    const [counter, setCounter ] = useState(value);

    const incrementa = () =>{
        setCounter(counter + 1);
        // setCounter((c)=>c+1);
    };

    const decrementa = () => {
        setCounter(counter - 1);
    }

    const resetValue = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CountApp</h1>
            <p>{counter}</p>
            <button onClick={incrementa}>+1</button>
            <button onClick={resetValue}>Reset</button>
            <button onClick={decrementa}>-1</button>
        </>
    );
};

CountApp.propTypes = {
    value: PropTypes.number.isRequired,
};

export default CountApp;