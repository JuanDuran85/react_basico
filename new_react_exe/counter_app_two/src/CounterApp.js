import { PropTypes } from 'prop-types';
import { React } from 'react';

const CounterApp = ({valor}) => {

    const clickBotton = (e) => {
        console.log('click', e);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{valor}</h2>
            <button onClick={clickBotton}>+1</button>
        </>
    )
}

CounterApp.propTypes = {
    valor: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    valor: 0
}

export default CounterApp;