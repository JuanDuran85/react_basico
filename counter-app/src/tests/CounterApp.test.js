import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import CounterApp from '../CounterApp';

describe('Pruebas en CounterApp', () => {
    let wrapper = shallow(<CounterApp value={100}/>);;
    
    beforeEach(() => {
        wrapper = shallow(<CounterApp value={100}/>);
    });

    test('prueba con snapshot', () => {
        expect(wrapper).toMatchSnapshot();
        const result = wrapper.find('p').text().trim();
        expect(result).toEqual('100');
    });
    
    test('incrementar con el boton +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const result = wrapper.find('p').text().trim();
        // console.log(boton1.html());
        expect(result).toBe('101');
    });

    test('incrementar con el boton -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const result = wrapper.find('p').text().trim();
        // console.log(boton1.html());
        expect(result).toBe('99');
    });

    test('reset con el boton', () => {
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const result = wrapper.find('p').text().trim();
        // console.log(boton1.html());
        expect(result).toBe('100');
    });
});