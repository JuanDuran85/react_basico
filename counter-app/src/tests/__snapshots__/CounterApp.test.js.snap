// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp prueba con snapshot 1`] = `
<Fragment>
  <h1>
    CountApp
  </h1>
  <p>
    100
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
